db/view/view_update_generator.cc

```diff
@@ -74,7 +74,7 @@ future<> view_update_generator::start() {
                                 tracing::trace_state_ptr ts,
                                 streamed_mutation::forwarding fwd_ms,
                                 mutation_reader::forwarding fwd_mr) {
-                        return ::make_restricted_range_sstable_reader(s, std::move(permit), std::move(ssts), pr, ps, pc, std::move(ts), fwd_ms, fwd_mr);
+                        return ::make_restricted_range_sstable_reader(s, std::move(permit), ssts->get(), pr, ps, pc, std::move(ts), fwd_ms, fwd_mr);
```

Comment:
The changes here (view_update_generator::start()) don't really modify the logic of the function, they were needed because the types of the objects we are working on changed. We are still creating a new sstable_set based on the list, but the sstable reader expects an sstable_set_snapshot, which we get by calling get().